.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,26 @@
+name: pre-commit
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # make sure python versions are consistent with those used in .circleci/config.yml
+        python-version: ['3.7.12', '3.8.12']
```

Comment:
nit: what about python 3.9?

Reply:
good catch. there is only one job using 3.9 so I missed it.